### ISSUE TITLE
Union + $PropertyType as object not working properly

### ISSUE THREAD

**Issue by hackhat** [facebook/flow#4851]
Reproduce at: https://flow.org/try/#0AQ4ejYAEDMBsD2B3AUKYAXAngBwKbABiCCwAvMAN6a54BcVAhgwM4YBOAlgHYDmAvsH4BuNKGz5gAIUbtyVGvgaUARnTZc+gkWJASCAVW6cE3ecVIAfabNG7gAY1NtgzACQAFdgnztsAFVoAHiMTbgAaYAByfSiAPnlKZmiWKJEgA

Code:

```
    // @flow
    type Foo = { type: {a: string} };
    type Bar = { type: {b:string} };
    type Union = Foo | Bar;

    const a:$PropertyType<Union, 'type'> = {a: 's'}; // errors
    const b:$PropertyType<Union, 'type'> = {b: 's'}; // errorr
    const c:$PropertyType<Union, 'type'> = {a: 's', b: 'ss'}; // not error
```

I think it should allow for `a` to be `{a: 'xxx'}` or `{b: 'xxx'}`, so `$PropertyType` joins all of the join instead of selecting one the items.

**Comment by swansontec** [comment]
This doesn't seem like a bug. You are always allowed to have extra properties:

```js
const x: {a: string} = {a: 's', b: 'ss'} // Works fine
```

What happens if you use exact object types (`{| a: string |}`) instead of regular object types? Do you still see the problem?

**Comment by hackhat** [comment]
`const x: {a: string} = {a: 's', b: 'ss'} // Works fine` that's what I'm saying as well. I'm saying that `const a:$PropertyType<Union, 'type'> = {a: 's'};` errors, when is shouldn't

**Comment by apsavin** [comment]
@hackhat fixed in 0.55
